src/model/SurfaceControlMovableInsulation.hpp

```diff
@@ -0,0 +1,127 @@
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_SURFACECONTROLMOVABLEINSULATION_HPP
+#define MODEL_SURFACECONTROLMOVABLEINSULATION_HPP
+
+#include "ModelAPI.hpp"
+#include "ModelObject.hpp"
+
+#include "Material.hpp"
+#include "Surface.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+class Material;
+class Schedule;
+
+namespace detail {
+
+  class SurfaceControlMovableInsulation_Impl;
+
+} // detail
+
+/** SurfaceControlMovableInsulation is a ModelObject that wraps the OpenStudio IDD object 'OS:SurfaceControl:MovableInsulation'. */
+class MODEL_API SurfaceControlMovableInsulation : public ModelObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit SurfaceControlMovableInsulation(const Surface& surface);
+
+  virtual ~SurfaceControlMovableInsulation() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  /** @name Getters */
+  //@{
+
+  boost::optional<std::string> insulationType() const;
+
+  Surface surface() const;
+
+  boost::optional<std::string> surfaceName() const;
+
+  boost::optional<Material> material() const;
+
+  boost::optional<Schedule> schedule() const;
+
+  //@}
+  /** @name Setters */
+  //@{
+
+  bool setInsulationType(const std::string& insulationType);
+
+  void resetInsulationType();
+
+  bool setSurface(const Surface& surface);
+
+  bool setMaterial(const Material& material);
+
+  void resetMaterial();
```

Comment:
If we decide to not allow it to be in a broken state, there should be no reset here either.